pkg/config/evaluation.go

```diff
@@ -194,6 +192,35 @@ func (s *StepCondition) evaluateV1(config StepConfig, utils piperutils.FileUtils
 	}
 }
 
+func checkConfigKeyV1(configKey string, config StepConfig) (bool, error) {
```

Comment:
What about doing this in a recursive way? I think this would be simpler & shorter code.

Reply:
Done. Could you take a look again?